### PR TITLE
dependencies: migrate `storage_sync` resources to `hashicorp/go-azure-sdk`

### PR DESCRIPTION
**Pull request by katbyte**

### CODE REVIEW

**Review by zezo2019 — APPROVED** [review]

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Blocked on https://github.com/hashicorp/go-azure-sdk/issues/498

**Review by mbfrahry — APPROVED** [review]
LGTM after merging with main!

**Pull request merged**
